src/cargo/core/resolver/dep_cache.rs

```diff
@@ -529,9 +532,10 @@ impl RequirementError {
                         None => ActivateError::Fatal(anyhow::format_err!(
                             "Package `{}` does not have feature `{}`. It has an optional dependency \
                              with that name, but that dependency uses the \"dep:\" \
-                             syntax in the features table, so it does not have an implicit feature with that name.",
+                             syntax in the features table, so it does not have an implicit feature with that name.{}",
                             summary.package_id(),
-                            feat
+                            feat,
+                            closest
                         )),
```

Comment:
This error message is already telling the user that something exists that could be mistaken for a feature but wasn't.  It is unlikely that a typo would happen here and it feels like the two messages don't compose well.

Reply:
Hopefully 875e1cfb7a151e38a41c113441cd1be41f303b6f makes this better